graphql_compiler/query_pagination/__init__.py

```diff
@@ -81,17 +82,32 @@ def paginate_query_ast(
     remainder_queries: Tuple[ASTWithParameters, ...] = tuple()
     advisories: Tuple[PaginationAdvisory, ...] = tuple()
 
+    # Split the query if we should and we can
     result_size = query_analysis.cardinality_estimate
     num_pages = _estimate_number_of_pages(
         query_analysis.query_string_with_parameters, result_size, page_size
     )
     if num_pages > 1:
         pagination_plan, advisories = get_pagination_plan(query_analysis, num_pages)
-        if pagination_plan.vertex_partitions:
-            page_query, remainder_query = split_into_page_query_and_remainder_query(
-                query_analysis, pagination_plan
+        if len(pagination_plan.vertex_partitions) == 0:
+            pass
+        elif len(pagination_plan.vertex_partitions) == 1:
+            parameter_generator = generate_parameters_for_vertex_partition(
```

Comment:
nit: mind extracting `pagination_plan.vertex_partitions[0]` somewhere here and assigning it to a local? The repeated `[0]` dereferences keep making me look back for evidence that `[0]` is defined. The code is correct  right now but I think this would make it easier for the reader to convince themselves of that.